src/puppetlabs/puppetdb/config.clj

```diff
@@ -375,3 +377,23 @@
       validate-vardir
       convert-config
       configure-catalog-debugging))
+
+(defprotocol DefaultedConfig
```

Comment:
This along with create-defaulted-config function below would be a good candidate for TK or a shared library, see it's usage in the config-service def below. /cc @KevinCorcoran @cprice404